### ISSUE TITLE
cpprest_compat breaks cURL header when building with Clang 14

### ISSUE THREAD

**Issue by lilianmoraru** [microsoft/cpprestsdk#1710]
`#define dllimport` inside [cpprest_compat.h](https://github.com/microsoft/cpprestsdk/blob/80444805898842f7e5c62b256337c0897eb254bb/Release/include/cpprest/details/cpprest_compat.h) breaks [curl.h](https://github.com/curl/curl/blob/bfbde883af33397943df68a3ae01847a634d33bf/include/curl/curl.h#L120) on Clang 14.

See minimized example here: [godbolt](https://godbolt.org/z/T6nxGY4cz)

The issue was initially [reported to LLVM](https://github.com/llvm/llvm-project/issues/53269).
The LLVM issue was closed. For the reason, see these comments: [comment1](https://github.com/llvm/llvm-project/issues/53269#issuecomment-1158813473) and [comment2](https://github.com/llvm/llvm-project/issues/53269#issuecomment-1161732686).

**Comment by barcharcraz** [comment]
Yeah I can see why they closed that one. This header is pretty horrible.

**Comment by laveur** [comment]
FYI this project is now dead. You should move away from cpprest ASAP. See the readme file.

**Comment by barcharcraz** [comment]
Indeed, we will fix any super critical bugs, but we highly, highly, highly recommend folks move to another library. We usually recommend libcurl.
